### PR TITLE
Fix some issues related to unreachability with panic, break and continue statements

### LINKED ISSUE #36923: No "unreachable code" error for unreachable code on break and continue combination
**Description:**
$title.

**Steps to reproduce:**
```ballerina
function fn(float? v) {
    int y = 0;

    while y > 2 {
        if v is () {
            break;
        } else {
            continue;
        }
        int _ = 1; // Unreachable, but no error.
    }
}
```

But when we explicitly check `else if v is float` (which is always true), the error is logged.

```ballerina
function fn(float? v) {
    int y = 0;

    while y > 2 {
        if v is () {
            break;
        } else if v is float {
            continue;
        }
        int _ = 1; // Unreachable, OK.
    }
}
```
**Affected Versions:**
2201.1.0

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
$title

Fixes #36923
And addresses review suggestions in #36925 PR

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java [hunk 1]
```diff
@@ -988,6 +1007,8 @@ public static class AnalyzerData {
         boolean failureHandled;
         boolean returnedWithinQuery;
         boolean skipFurtherAnalysisInUnreachableBlock;
+        boolean hasPanicStatement;
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java:1010]: How is this different to errorThrown?

Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java:1010]: Can't we use the same? We seem to have multiple fields and I'm wondering if we can probably clean these up.

Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java]: `errorThrown` had been used for on fail clause related scenarios. Will check whether we can clean these up. Created https://github.com/ballerina-platform/ballerina-lang/issues/36951 to track this.

**Comment by MaryamZi** [comment]
@lasinicl, please share the PR to master with these changes too?

**Comment by lasinicl** [comment]
PR for master: https://github.com/ballerina-platform/ballerina-lang/pull/37013

**Pull request merged**